src/main/java/Duke/Duke.java

```diff
@@ -0,0 +1,359 @@
+
+        //create new Event task
+        tasksArrayList.add(new Event(taskName, dueDate));
+        taskCounter++;
+    }
+
+    private static String getEventTaskName(String input, int slashIndex) throws DukeException {
+        final int EVENT_WORD_LENGTH = 6;
+        int taskNameLastIndex = slashIndex - 1; //the last index of the task name
+
+        //if the slash is not present in the input
+        if (slashIndex == -1) {
+            throw new DukeException("☹ OOPS!!! The description of an event task requires a task name" + System.lineSeparator()
+                    + "\tfollowed by a front slash, and then a specific time after." + System.lineSeparator()
+                    + "\ti.e. event team project meeting /on 2-10-2019 2-4pm");
+        } else if (taskNameLastIndex <= EVENT_WORD_LENGTH) { //if the slash exists but the task name is empty
+            throw new DukeException("☹ OOPS!!! The name of an event task cannot be empty.");
+        }
+
+        String taskName = input.substring(EVENT_WORD_LENGTH, taskNameLastIndex);
+
+        //if taskName are just filled with whitespaces
+        if (taskName.replace(" ", "").isEmpty()) {
+            throw new DukeException("☹ OOPS!!! The name of an event task cannot be empty.");
+        }
+        return taskName;
+    }
+
+    private static void addDeadlineTask(String input) throws DukeException {
+
+        //get the name of the task
+        int slashIndex = input.indexOf('/');
+        String taskName = getDeadlineTaskName(input, slashIndex);
+
+        //get the due date of the task
+        String dueDate = getDueDate(input, slashIndex);
+
+        //create new Deadline task
+        tasksArrayList.add(new Deadline(taskName, dueDate));
+
+        taskCounter++;
+    }
+
+    private static String getDeadlineTaskName(String input, int slashIndex) throws DukeException {
+        final int DEADLINE_WORD_LENGTH = 9;
+        int taskNameLastIndex = slashIndex - 1; //the last index of the tsak name
+
+        //if the slash is not present in the input
+        if (slashIndex == -1) {
+            throw new DukeException("☹ OOPS!!! The description of an deadline task requires a task name" + System.lineSeparator()
+                    + "\tfollowed by a front slash, and then a specific date/time after." + System.lineSeparator()
+                    + "\ti.e. deadline submit report /by 11-10-2019 5pm");
+        } else if (taskNameLastIndex <= DEADLINE_WORD_LENGTH) { //if the slash exists but the task name is empty
+            throw new DukeException("☹ OOPS!!! The name of an deadline task cannot be empty.");
+        }
+
+        String taskName = input.substring(DEADLINE_WORD_LENGTH, taskNameLastIndex);
+
+        //if taskName are just filled with whitespaces
+        if (taskName.replace(" ", "").isEmpty()) {
+            throw new DukeException("☹ OOPS!!! The name of a deadline task cannot be empty.");
+        }
+        return taskName;
```

Comment:
You could [keep things more simple](https://nus-cs2113-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#practice-kissing) by making the [normal execution path more prominent](https://nus-cs2113-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#make-the-happy-path-prominent). 

E.g. Start with what steps are done when there is a valid input.